src/utils.ts

```diff
@@ -172,7 +173,8 @@ export async function querySubgraph(
     for (const e of json.errors) {
       console.error("querySubgraph:", e.message);
     }
-    return null;
+    if (intent < 3) querySubgraph(url, query, variables, intent += 1);
```

Comment:
You can just do `intent + 1` here, no need to mutate it!

Also perhaps we call this `retries` to be clearer?

Reply:
Actually hmm, it would be nicer if the count was decremented. This would allow the caller to choose the retry count, and you'd default it to 3, ie. `retries = 3` in the function signature.

Reply:
Yeah sounds even better :+1: